runtime/src/main/java/org/corfudb/runtime/collections/streaming/StreamPollingScheduler.java

```diff
@@ -0,0 +1,321 @@
+                pollBatchSize);
+        Preconditions.checkArgument(pollThreshold > 1, "pollThreshold=%s has to be > 1",
+                pollThreshold);
+        Preconditions.checkArgument(pollPeriod.toMillis() > 1, "pollPeriod=%s has to be > 1ms",
+                pollPeriod.toMillis());
+
+        this.scheduler = scheduler;
+        this.workers = workers;
+        this.pollPeriod = pollPeriod;
+        this.pollBatchSize = pollBatchSize;
+        this.pollThreshold = pollThreshold;
+        this.runtime = runtime;
+        this.sequencerView = runtime.getSequencerView();
+        tick();
+    }
+
+    public class Tick implements Runnable {
+        @Override
+        public void run() {
+            try {
+                schedule();
+            } catch (InterruptedException ie) {
+                log.warn("Scheduler thread has been interrupted!", ie);
+                Thread.currentThread().interrupt();
+            } catch (Throwable t) {
+                log.error("Unexpected throwable!", t);
+            }
+        }
+    }
+
+    private void tick() {
+        scheduler.submit(new Tick());
+    }
+
+    public void addTask(@Nonnull StreamListener streamListener, @Nonnull String namespace,
+                        @Nonnull String streamTag, @Nonnull List<String> tablesOfInterest,
+                        long lastAddress, int bufferSize) {
+
+        Preconditions.checkArgument(bufferSize >= pollThreshold);
+        synchronized (allTasks) {
+            if (allTasks.containsKey(streamListener)) {
+                // Multiple subscribers subscribing to same namespace and table is allowed
+                // as long as the hashcode() and equals() method of the listeners are different.
+                throw new StreamingException(
+                        "StreamingManager::subscribe: listener already registered " + streamListener);
+            }
+            StreamingTask task = new StreamingTask(runtime, workers, namespace, streamTag, streamListener,
+                    tablesOfInterest, lastAddress, bufferSize);
+            allTasks.put(streamListener, task);
+            log.debug("addTask: added {} {} {}", streamListener, namespace, streamTag);
+            allTasks.notifyAll();
+        }
+    }
+
+    public void removeTask(@Nonnull StreamListener streamListener) {
+        synchronized (allTasks) {
+            allTasks.remove(streamListener);
+            allTasks.notifyAll();
+        }
+    }
+
+    // hole filling mitigation
+    // sequencer regression mitigation
+    // fairness?
+
+    private void waitForTasks() throws InterruptedException {
+        synchronized (allTasks) {
+            while (allTasks.isEmpty()) {
+                allTasks.wait();
+            }
+        }
+    }
+
+    private List<StreamingTask> getTasks(StreamStatus status, Predicate<StreamingTask> predicate) {
+        List<StreamingTask> result = new ArrayList<>();
```

Comment:
`return allTasks.values().stream().filter(predicate.and(task -> task.getStatus() == status)).collect(Collectors.toList());`
just a suggestion

Reply:
Done.